src/agent/onefuzz/src/asan.rs

```diff
@@ -20,7 +20,11 @@ pub struct AsanLog {
 
 impl AsanLog {
     pub fn parse(text: String) -> Option<Self> {
-        let (summary, sanitizer, fault_type) = parse_summary(&text)?;
+        let (summary, sanitizer, fault_type) = match parse_summary(&text) {
+            Some(x) => x,
+            None => parse_failure(&text)?,
```

Comment:
In falling back to `parse_failure()`, aren't we throwing away detail?

Should `parse()` really be returning an enum that explicitly encodes the different cases of:
1. We parsed an ASAN log where the _target_ tripped a fault oracle
2. We parsed an ASAN log where there was an internal ASAN error
3. There was no ASAN-specific logging at all

Reply:
Discussed offline.